libsolidity/ast/ASTUtils.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <libsolidity/ast/AST.h>
+#include <libsolidity/ast/ASTUtils.h>
+
+namespace dev
+{
+namespace solidity
+{
+
+VariableDeclaration const* rootVariableDeclaration(VariableDeclaration const* varDecl)
+{
+	Identifier const* identifier;
+	VariableDeclaration const* referencedVarDecl;
+	while ((identifier = dynamic_cast<Identifier const*>(varDecl->value().get())) &&
```

Comment:
Can you maybe refactor this a little and extract the statements from the loop condition? Maybe a do-while loop would be better.